### ISSUE TITLE
Incorrect warning when md is too long

### ISSUE THREAD

**Issue by 1298345095** [DavidAnson/markdownlint#1691]
When text is short, there is no problem (split into multiple files)

Such as this:
````plaintext
```c
/* header files go up here */
/* note that C comments are enclosed within a slash and
 a star, and may wrap over lines */
// two slashes work too (and may be preferred)
#include <stdio.h>
// main returns an integer
int main(int argc, char *argv[]) {
    /* printf is our output function;
    by default, writes to standard out */
    /* printf returns an integer, but we ignore that */
    printf("hello, world\n");
    /* return 0 to indicate all went well */
    return (0);
}
```
````

Warning: MD018/no-missing-space-atx: No space after hash on atx style heading (Mistakenly identify \#include as title)

And this:
````plaintext
```makefile
hw: hw.o helper.o
	gcc -o hw hw.o helper.o -lm
hw.o: hw.c
	gcc -O -Wall -c hw.c
helper.o: helper.c
	gcc -O -Wall -c helper.c
clean:
	rm -f hw.o helper.o hw
```
````

MD010/no-hard-tabs: Hard tabs [Column: 1]

I already set "MD010":{
            "ignore_code_languages": ["makefile"]
        },

**Comment by DavidAnson** [comment]
I don't understand what problem you are reporting - your scenario and example produce no warnings in the demo app:
<https://dlaa.me/markdownlint/#%25m%23%20Issue%201691%0A%0A%60%60%60makefile%0Ahw%3A%20hw.o%20helper.o%0A%09gcc%20-o%20hw%20hw.o%20helper.o%20-lm%0Ahw.o%3A%20hw.c%0A%09gcc%20-O%20-Wall%20-c%20hw.c%0Ahelper.o%3A%20helper.c%0A%09gcc%20-O%20-Wall%20-c%20helper.c%0Aclean%3A%0A%09rm%20-f%20hw.o%20helper.o%20hw%0A%60%60%60%0A%0A%3C!--%20markdownlint-configure-file%20%7B%0A%20%20%22no-hard-tabs%22%3A%20%7B%0A%20%20%20%20%22ignore_code_languages%22%3A%20%5B%0A%20%20%20%20%20%20%22makefile%22%0A%20%20%20%20%5D%0A%20%20%7D%0A%7D%20--%3E%0A>

**Comment by 1298345095** [comment]
> I don't understand what problem you are reporting - your scenario and example produce no warnings in the demo app: https://dlaa.me/markdownlint/#%25m%23%20Issue%201691%0A%0A%60%60%60makefile%0Ahw%3A%20hw.o%20helper.o%0A%09gcc%20-o%20hw%20hw.o%20helper.o%20-lm%0Ahw.o%3A%20hw.c%0A%09gcc%20-O%20-Wall%20-c%20hw.c%0Ahelper.o%3A%20helper.c%0A%09gcc%20-O%20-Wall%20-c%20helper.c%0Aclean%3A%0A%09rm%20-f%20hw.o%20helper.o%20hw%0A%60%60%60%0A%0A%3C!--%20markdownlint-configure-file%20%7B%0A%20%20%22no-hard-tabs%22%3A%20%7B%0A%20%20%20%20%22ignore_code_languages%22%3A%20%5B%0A%20%20%20%20%20%20%22makefile%22%0A%20%20%20%20%5D%0A%20%20%7D%0A%7D%20--%3E%0A

<img width="854" height="520" alt="Image" src="https://github.com/user-attachments/assets/98c7caea-33df-45a0-87b3-52a5f74c2a6a" />

**Comment by DavidAnson** [comment]
It looks like you quoted my own message back to me. Can you please demonstrate the problem, either in the demo page or by pointing to a repository or gist?

**Comment by 1298345095** [comment]
> It looks like you quoted my own message back to me. Can you please demonstrate the problem, either in the demo page or by pointing to a repository or gist?

No, I took the screenshot to show the line numbers, indicating that the issue occurs when there are many lines.

**Comment by 1298345095** [comment]
> It looks like you quoted my own message back to me. Can you please demonstrate the problem, either in the demo page or by pointing to a repository or gist?

You can create a markdown file with many lines to demonstrate the problem. When I finish the markdown, I will upload it to my repository.
